java-checks-test-sources/src/main/java/checks/regex/ReluctantQuantifierWithEmptyContinuationCheck.java

```diff
@@ -7,25 +7,48 @@
 
 public class ReluctantQuantifierWithEmptyContinuationCheck {
 
-  @Email(regexp = ".*?") // Noncompliant {{Remove the '?' from this unnecessarily reluctant quantifier.}}
   void noncompliant(String str) {
-    Pattern.compile(".*?x?").matcher(str).find(); // Noncompliant [[sc=22;ec=25]] {{Fix this reluctant quantifier that will only ever match the empty string.}}
-    Pattern.compile(".*?$").matcher(str).find(); // Noncompliant {{Remove the '?' from this unnecessarily reluctant quantifier.}}
-    Pattern.compile(".*?x?^").matcher(str).find(); // Noncompliant {{Fix this reluctant quantifier that will only ever match the empty string.}}
-    str.split(".*?x?^"); // Noncompliant {{Fix this reluctant quantifier that will only ever match the empty string.}}
+    // ====== Reluctant quantifier that will only ever match the empty string ======
+    // Partial match
+    Pattern.compile(".*?").matcher(str).find(); // Noncompliant {{Fix this reluctant quantifier that will only ever match 0 repetitions.}}
+    Pattern.compile(".+?").matcher(str).find(); // Noncompliant {{Fix this reluctant quantifier that will only ever match 1 repetition.}}
+    Pattern.compile(".{4}?").matcher(str).find(); // Noncompliant {{Fix this reluctant quantifier that will only ever match 4 repetitions.}}
```

Comment:
The message has changed and is now reporting the number of repetitions instead of "empty string". It may sound strange when the value is "0", but is more accurate since the previous message was erroneous for the cases != 0,